### PR TITLE
FA-98, basics of the legacy app, runnable

### PR DESCRIPTION
**Pull request by HamdanSCode**
May need to move some things lower down into a legacyappcore.

### CODE REVIEW

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    private var errorId: Int? = null
+    private var firstName: Entry = Entry("", LoginError.NameError.None)
+    private var lastName: Entry = Entry("", LoginError.NameError.None)
+    private var birthDay: Entry = Entry("", LoginError.BirthDateError.None)
+    private var email: Entry = Entry("", LoginError.EmailError.None)
+    private var location: Entry = Entry("", LoginError.LocationError.None)
+    private var isDateDialogOpen: Boolean = false
+    private var isLoading: Boolean = false
+
+    var onCreateAccountUpdate: ((CreateAccountState) -> Unit)? = null
```
Comment (ForzenAcademy) [inline applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt:47]: Why make this a lambda?

Comment (ForzenAcademy) [reply applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt:47]: The intent here is that we're getting away from lambdas, the "easy" update method, and going to MutableStateFlow as is the expected company practice

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyLoginViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    private var email: Entry = Entry("", LoginError.EmailError.None)
+    private var code: Entry = Entry("", LoginError.CodeError.None)
+    private var showInfoDialog: Boolean = false
+    private var inputtingCode: Boolean = false
+    private var isLoading: Boolean = false
+    private var hasError: Boolean = false
+
+    var onLoginUpdate: ((LoginState) -> Unit)? = null
```
Comment (ForzenAcademy) [inline applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyLoginViewModel.kt:45]: StateFlow

File: applegacy/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="button_small_1">40dp</dimen>
```
Comment (ForzenAcademy) [inline applegacy/src/main/res/values/dimens.xml:3]: Dimen names should be camel case

File: applegacy/src/main/res/values/attrs.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <attr name="legacy_background" format="color" />
```
Comment (ForzenAcademy) [inline applegacy/src/main/res/values/attrs.xml:3]: Can make these camel case

File: core/build.gradle [hunk 5]
```diff
@@ -35,7 +35,7 @@ dependencies {
 
     implementation 'androidx.core:core-ktx:1.9.0'
     implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.5.1'
-    implementation 'androidx.activity:activity-compose:1.7.0-beta01'
+    implementation 'androidx.activity:activity-compose:1.6.1'
```
Comment (ForzenAcademy) [inline core/build.gradle:38]: Beta no bueno?

Comment (HamdanSCode) [reply core/build.gradle:38]: It was a leftover from when I was cleaning up, didn't see a point in it being in its own pr as a small thing

**Pull request merged**